packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java

```diff
@@ -414,6 +414,37 @@ public void onComplete(
           break;
         }
 
+      case "OnDisconnect#set":
+        {
+          Object value = arguments.get("value");
+          Object priority = arguments.get("priority");
+          DatabaseReference reference = getReference(database, arguments);
+          if (priority != null) {
+            reference
+                .onDisconnect()
+                .setValue(value, (double) priority, new DefaultCompletionListener(result));
```

Comment:
Since this can be either a String or a double we should not cast this to double without checking.

Reply:
Correct.
At some point I tried to take a similar approach as in `DataReference.setValue`.
Priority argument comes as dynamic from dart and mapped as Object to java.
Now the difference is in the android API: `OnDisconnect.setValue` holds 3 signatures for the priority argument, being: double, Map and String, whereas `DataReference.setValue` is more generic and exposes a single signature with Object. Will update to 
```
if (priority instanceof String) {
  reference.onDisconnect()
    .setValue(value, (String) priority, new DefaultCompletionListener(result));
} else if (priority instanceof Double) {
  reference.onDisconnect()
    .setValue(value, (double) priority, new DefaultCompletionListener(result));
} else if (priority instanceof Map) {
  reference.onDisconnect()
    .setValue(value, (Map) priority, new DefaultCompletionListener(result));
}
```